bindings/c/include/types.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+#include <stddef.h>
+
+typedef struct ManifoldManifold ManifoldManifold;
+typedef struct ManifoldSimplePolygon ManifoldSimplePolygon;
+typedef struct ManifoldPolygons ManifoldPolygons;
+typedef struct ManifoldMesh ManifoldMesh;
+typedef struct ManifoldMeshGL ManifoldMeshGL;
+typedef struct ManifoldCurvature ManifoldCurvature;
+typedef struct ManifoldComponents ManifoldComponents;
+typedef struct ManifoldMeshRelation ManifoldMeshRelation;
+typedef struct ManifoldBox ManifoldBox;
+typedef struct ManifoldMaterial ManifoldMaterial;
+typedef struct ManifoldExportOptions ManifoldExportOptions;
+
+typedef struct ManifoldManifoldPair {
+  ManifoldManifold* first;
+  ManifoldManifold* second;
+} ManifoldManifoldPair;
+
+typedef struct ManifoldVec2 {
+  float x;
+  float y;
+} ManifoldVec2;
+
+typedef struct ManifoldVec3 {
+  float x;
+  float y;
+  float z;
+} ManifoldVec3;
```

Comment:
I take it this is because `glm` isn't a c library?

Reply:
Yes it isn't, there is a `cglm` counterpart library. This way there isn't any dependency on `glm` to be able to understand the structs (at least that was my assumption). Not sure if there is a better way to handle the passing of vector arrays back and forth, I might have taken an overly cautious route.

Reply:
Up to you as the user. I find glm to be an excellent library that makes my life much easier. If you want to include cglm for these bindings that's fine, but I don't have a strong opinion.

Reply:
Okay. Since there isn't actually any number crunching happening and it's just a struct of floats I think this way works well enough then.